chrome/content/zotero/preferences/preferences_sync.jsx

```diff
@@ -831,21 +831,24 @@ Zotero_Preferences.Sync = {
 				}
 				break;*/
 			
-			case 'restore-to-server':
-				var buttonFlags = (ps.BUTTON_POS_0) * (ps.BUTTON_TITLE_IS_STRING)
-					+ (ps.BUTTON_POS_1) * (ps.BUTTON_TITLE_CANCEL)
-					+ ps.BUTTON_POS_1_DEFAULT;
-				var index = ps.confirmEx(
-					null,
-					Zotero.getString('general.warning'),
-					Zotero.getString(
-						'zotero.preferences.sync.reset.restoreToServer',
-						[Zotero.clientName, library.name, ZOTERO_CONFIG.DOMAIN_NAME]
-					),
-					buttonFlags,
-					Zotero.getString('zotero.preferences.sync.reset.restoreToServer.button'),
-					null, null, null, {}
-				);
+			case 'restore-to-server': {
+				let apiKey = await Zotero.Sync.Data.Local.getAPIKey();
+				let client = Zotero.Sync.Runner.getAPIClient({ apiKey });
+				var keyInfo = await Zotero.Sync.Runner.checkAccess(client, { timeout: 5000 });
+				let { keys: remoteKeysArray } = await client.getKeys('user', keyInfo.userID, { target: 'items/top' });
+				let remoteKeys = new Set(remoteKeysArray);
+				let localKeys = new Set(await Zotero.Items.getAll(Zotero.Libraries.userLibraryID, true, false, false, true));
```

Comment:
This is using `onlyTopLevel`, so in theory you could be wiping out hundreds of attachments, notes, and/or annotations. Probably too confusing to count annotations as "items", but we should probably at least count attachments and notes. Can just omit `asIDs` (contra my other comment), filter to `isRegularItem() || isNote() || isFileAttachment()`, and map to`.key`.

We could also show annotations separately, but that's probably getting too complicated, since the main use case here is just preventing wiping out a whole library.